### PR TITLE
Add EIP: EVM Modular Arithmetic Extensions

### PR DESCRIPTION
**Pull request by jwasinger**

### CODE REVIEW

File: EIPS/eip-8743.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+eip: 8743
```
Comment (abcoathup) [inline EIPS/eip-8743.md:2]:
```suggestion
eip: 7747
```
Assigning next sequential EIP/ERC/RIP number. Numbers are no longer the issue number.

Please also update the filename.

File: EIPS/eip-8743.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+title: EVM Modular Arithmetic Extensions
+status: Draft
+type: standards track
```
Comment (abcoathup) [inline EIPS/eip-8743.md:5]:
```suggestion
type: Standards Track
```

File: EIPS/eip-8743.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+author: Jared Wasinger <@jwasinger>, Alex Beregszaszi (@axic)
```
Comment (abcoathup) [inline EIPS/eip-8743.md:6]:
```suggestion
author: Jared Wasinger (@jwasinger), Alex Beregszaszi (@axic)
```

File: EIPS/eip-8743.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+discussions-to:
```
Comment (abcoathup) [inline EIPS/eip-8743.md:7]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7747-evm-modular-arithmetic-extensions/20608
```

File: EIPS/eip-7747.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+* Assert that the destination of the copy falls entirely within EVM memory.
+* Charge `gas_mulmodx(active_context.element_size / 8) * count` and copy the values into EVM memory starting at offset `dest`, laying them out in big-endian ordering, padded such that size of each is a multiple of 64 bits.
+
+#### `STOREX(0xc2)`
+
+**Stack in**: `dest source count`
+
+**Stack out**: none
+
+**Description**: copies values layed out in EVM memory as 64bit-aligned big-endian numbers into a contiguous section of the currently-active field context's value space.
```
Comment (mratsim) [inline EIPS/eip-7747.md:157]: It is unclear if the layout is compatible with BLS12-381 (i.e. EIP-2537) layout, I think it is but padding should be mentioned

File: EIPS/eip-7747.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+### Data Transfer
+
+#### `LOADX(0xc1)`
+
+**Stack in**: `(top of stack) dest source count`
+
+**Stack out**: none
+
+**Description**: copies a number of contiguous values in the currently-active field context value space into EVM memory.
```
Comment (mratsim) [inline EIPS/eip-7747.md:142]: Padding to next multiple of 32 bytes should be mentioned

Comment (jwasinger) [reply EIPS/eip-7747.md]: I've added a note clarifying the EVM memory serialization format for values:  big-endian padding until the length is the same as the length of the modulus padded to the nearest 64 bits.

**Comment by jwasinger** [comment]
We are going to stick with EIP 6690 and push further changes to that EIP.

**Pull request closed without merging**